support-frontend/assets/helpers/abTests/abtest.ts

```diff
@@ -251,44 +366,45 @@ const trackOptimizeExperiment = (
 	);
 };
 
+type VariantAssignment =
+	| { type: 'NOT_ASSIGNED' }
+	| { type: 'ASSIGNED'; variantIndex: number };
+
```

Comment:
Again, I've updated this function to return a tagged union. It's mainly to remove the magic `-1` we had before when it just returned a `number`. It's maybe a little overkill but I like the uniformity with `getUserParticipation`